### PR TITLE
[#1092] Add draggable getter/setter

### PR DESCRIPTION
**Pull request by MaanasArora**
## Changes:

- drafted ex.Actor.draggable utility getter/setter

### CODE REVIEW

**Comment by kamranayub** [comment]
Thank you @MaanasArora, nice start! I was wondering @eonarheim if you foresee any issues adding handlers during the set operation. One thing I think we'd want to make sure to do is remove the handlers when `draggable` is unset.

Is this something ECS will help with, would `Draggable` be a component that just gets added/removed? I don't know enough to say right now but that's what came to mind.

**Comment by eonarheim** [comment]
Thanks @MaanasArora this is a good start! I agree, we should be sure that the handlers are removed when unset, and that multiple handlers cannot be accidentally set if `.draggable = true` is called multiple times.

@kamranayub I think you are on the right track. For now I don't think we should consider an ECS implementation, we'll likely want to look into a `Draggable` Component/System pair when we have a a more solid ECS foundation.

File: src/engine/Actor.ts [hunk 1]
```diff
@@ -527,6 +527,48 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
...
+  public get draggable(): boolean {
+    return this._draggable;
+  }
+
+  public set draggable(isDraggable: boolean) {
+    if (isDraggable) {
+      this._draggable = true;
+
+      if (isDraggable) {
+        this.collisionType = CollisionType.PreventCollision;
```
Comment (eonarheim) [inline src/engine/Actor.ts]: Setting the collision type to `PreventCollision` when `draggable = true` may be unexpected for users of Excalibur, I vote we do not set collision type. Thoughts @kamranayub?

**Comment by MaanasArora** [comment]
Thanks, @kamranayub @eonarheim! 😃 I will update the PR with the edits you pointed out:

- remove handlers when `draggable` is unset
- ensure that multiple handlers are not accidentally added (by making sure `draggable` was not already set before adding handlers)
- remove the line that sets `collisionType` to `PreventCollision`

**Comment by MaanasArora** [comment]
I have committed my changes :)

File: src/engine/Actor.ts [hunk 2]
```diff
@@ -527,6 +527,64 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
...
+      } else if (!isDraggable && this._draggable) {
+        this.off('pointerdragstart', () => {
+          this._dragging = true;
+        });
+
+        this.off('pointerdragstart', () => {
+          this._dragging = false;
+        });
+
+        this.off('pointerdragmove', (pe: PointerEvent) => {
```
Comment (eonarheim) [inline src/engine/Actor.ts]: @MaanasArora Last request for this PR, thanks for working so hard on this. In order for `this.off(...)` to work correctly it needs to be based the same function reference that was used in the `this.on(...)`. We'll need to store those somewhere, perhaps as private methods on actor?

Something like
```typescript
private _pointerDragStartHandler = () => {
   this._dragging = true;
}
...

 public set draggable(isDraggable: boolean) {
    if (isDraggable) {
      if (isDraggable && !this._draggable) {
        this.on('pointerdragstart', this._pointerDragStartHandler );
      ...
      } else if (!isDraggable && this._draggable) {
        this.off('pointerdragstart', this._pointerDragStartHandler );
      ...
}

**Comment by MaanasArora** [comment]
@eonarheim No problem, sorry for the error, and thank you for bringing it to attention! I followed your suggestion to store the functions as private methods and committed the changes.

**Comment by MaanasArora** [comment]
Alright, thanks! I have marked the PR as ready for review.

**Comment by eonarheim** [comment]
@MaanasArora Thanks again!

**Pull request merged**